src/bsoncxx/document/value.cpp

```diff
@@ -53,6 +53,9 @@ value& value::operator=(const value& rhs) {
     return *this;
 }
 
+value::value(value&& rhs) noexcept: _data{std::move(rhs._data)}, _length{std::move(rhs._length)} {
```

Comment:
I think I would like you to revert this part of the change (just the value move constructor), and add -Wno-error=maybe-uninitialized to the ASAN build.